### PR TITLE
[GOBBLIN-806]Enable metrics reporter during dataset discovery for retention job

### PR DESCRIPTION
**Pull request by ZihanLi58**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-806

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
 1. Change datasetCleaner to pass the eventSubmitter to dataset finder
 2. Change MultiDatasetFinder to enable datasetFinder with/without EventSubmitter

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Test on faro for both ConfigBasedCleanabledDatasetFinder and TMSBasedCleanableDatasetsFinder for retention job.

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by ZihanLi58** [comment]
@sv2000 Can you take a look at this?

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/profile/MultiCleanableDatasetFinder.java [hunk 1]
```diff
@@ -52,6 +53,10 @@ public MultiCleanableDatasetFinder(FileSystem fs, Properties jobProps) {
     super(fs, jobProps);
```
Comment (sv2000) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/profile/MultiCleanableDatasetFinder.java]: Should we create a default eventSubmitter here and call MultiCleanableDatasetFinder() which accepts eventSubmitter as an argument?

Comment (ZihanLi58) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/profile/MultiCleanableDatasetFinder.java]: In the super class MultiDatasetFinder, the construction MultiDatasetFinder(fs, jobProps) will create a default eventSubmitter and call MultiDatasetFinder() which accepts eventSubmitter as an argument. So I think there is no need to create a default eventSubmitter here.

Comment (sv2000) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/profile/MultiCleanableDatasetFinder.java]: hmm..ok. Can't we change the super class constructor to MultiDatasetFinder(fs, jobProps, eventSubmitter) and create the submitter in the MultiCleanableDatasetFinder() constructor and pass it to the parent class?

**Pull request closed without merging**